tests/console/commands/test_init.py

```diff
@@ -606,6 +606,7 @@ def test_interactive_with_wrong_dependency_inputs(
         "MIT",  # License
         "^3.8",  # Python
         "",  # Interactive packages
+        "foo 1.19.2",
```

Comment:
what is going on with this testcase?  why is "foo 1.19.2" valid but "pendulum 2.0.0" is invalid?  why do is "pendulum 2.0.0" tested three times?

Reply:
Yeah, now it's bugging me.
This changes was add in https://github.com/python-poetry/poetry/pull/5838 with by @mkniewallner with @neersighted review. Maybe one of them could bring us some understanding.

Reply:
Another point is that the `foo` package is not added in repo.add_package above. And testing locally with `repo._packages.clear()` before testing seems to not cause effect. Is it the `TestRepository` being considered at all?

Reply:
Looks like a simple oversight, please feel free to clean it up.